OpTestLogger.py

```diff
@@ -0,0 +1,124 @@
+
+# This implements all the python logger setup for op-test
+
+import os
+import sys
+from datetime import datetime
+import logging
+from logging.handlers import RotatingFileHandler
+
+class OpTestLogger():
+    '''
+    This class is used as the main global logger and handler initialization module.
+
+    See common/HelloWorld.py as an example for the usage within an op-test module.
+    '''
+    def __init__(self):
+        '''
+        Provide defaults and setup the minimal StreamHandlers
+        '''
+        self.parent_logger = 'op-test'
+        self.maxBytes_logger_file = 2000000
+        self.maxBytes_logger_debug_file = 2000000
+        self.backupCount_logger_files = 5
+        self.backupCount_debug_files = 5
+        self.logdir = os.getcwd()
+        self.logger_file = 'main.log'
+        self.logger_debug_file = 'debug.log'
+        self.optest_logger = logging.getLogger(self.parent_logger)
+        # clear the logger of any handlers in this shell
+        self.optest_logger.handlers = []
+        # need to set the logger to deepest level, handlers will filter
+        self.optest_logger.setLevel(logging.DEBUG)
+        # we need to init stream handler as special case to keep all happy
+        self.sh = logging.StreamHandler()
+        # save the sh level for later refreshes
+        self.sh_level = logging.ERROR
+        self.sh.setLevel(self.sh_level)
+        self.sh.setFormatter(logging.Formatter('\n%(asctime)s:%(name)s:%(levelname)s:%(message)s'))
```

Comment:
logger already adds one new line, this new line is not necessary. For each log message one extra blank line we are adding, that is not necessary,
```
2018-02-18 18:10:05,290:op-test.testcases.HelloWorld:DEBUG:HelloWorld runTest debug call

2018-02-18 18:10:05,291:op-test:DEBUG:Back from XMLTestRunner

2018-02-18 18:10:05,291:op-test:INFO:Exit with Result errors="0" and failures="0"

2018-02-18 18:10:05,291:op-test:INFO:See Main Log File for Results
```

Reply:
Done